### ISSUE TITLE
Unreachable throw statement doesn't cause a warning.

### ISSUE THREAD

**Issue by MarkPflug** [dotnet/roslyn#9138]
The following minimal program doesn't produce a warning about the throw statement being unreachable. This is in VS2015 update 1, and the behavior is the same in VS2013.

```
class Program {
    static void Main() {
        return;     
        throw new System.Exception(); // no warning
    }
}
```

Any other statement that I've tried instead of throw will produce a warning.

Apparently this is an old issue:
http://stackoverflow.com/questions/6371564/why-does-throwing-2-exceptions-in-a-row-not-generate-an-unreachable-code-warning

**Comment by gafter** [comment]
We may be able to fix this only in "strict" mode, as adding this warning could break some customer builds.

**Comment by gafter** [comment]
@MadsTorgersen Can we specify that we don't warn on unreachable blocks, empty statements `;`, and throw statements? We intentionally avoid giving these diagnostics in the compiler for compatibility, and there is a reasonable argument that this is the most useful thing for the programmer.

**Comment by gafter** [comment]
Recorded at https://github.com/ECMA-TC49-TG2/spec/issues/956 for possible revision in the next iteration of the ECMA C# language specification (i.e. to specify that those three statements do not deserve an unreachable diagnostic).
